src/main/java/com/hubspot/jinjava/el/ext/CollectionMembershipOperator.java

```diff
@@ -7,39 +7,43 @@
 import de.odysseus.el.tree.impl.ast.AstBinary;
 import de.odysseus.el.tree.impl.ast.AstBinary.SimpleOperator;
 import de.odysseus.el.tree.impl.ast.AstNode;
-import java.util.Collection;
 import java.util.Objects;
 import javax.el.ELException;
 import org.apache.commons.lang3.StringUtils;
 
 public class CollectionMembershipOperator extends SimpleOperator {
 
   @Override
-  public Object apply(TypeConverter converter, Object o1, Object o2) {
-    if (o2 == null) {
+  public Object apply(TypeConverter converter, Object value, Object maybeIterable) {
+    if (maybeIterable == null) {
       return Boolean.FALSE;
     }
 
-    if (CharSequence.class.isAssignableFrom(o2.getClass())) {
-      return StringUtils.contains((CharSequence) o2, Objects.toString(o1, ""));
+    if (CharSequence.class.isAssignableFrom(maybeIterable.getClass())) {
+      return StringUtils.contains(
+        (CharSequence) maybeIterable,
+        Objects.toString(value, "")
+      );
     }
 
-    if (Collection.class.isAssignableFrom(o2.getClass())) {
-      Collection<?> collection = (Collection<?>) o2;
-
-      for (Object value : collection) {
-        if (value == null) {
-          if (o1 == null) {
-            return Boolean.TRUE;
+    if (maybeIterable instanceof Iterable) {
+      for (Object element : (Iterable) maybeIterable) {
+        if (element == null) {
+          if (value == null) {
+            return true;
           }
         } else {
           try {
-            return collection.contains(converter.convert(o1, value.getClass()));
```

Comment:
What we had before was that we would cast the value to the type of the first element of the collection, and then return on that result. So `2 is in ['one', 2]` would fail because `2` gets casted to `'2'`.

Reply:
Can you check the performance of this? I believe I did this back in https://github.com/HubSpot/jinjava/pull/314/files for performance reasons but would be good to double check. We could also split this into a separate PR to be safe.

Reply:
There is a slight performance hit. For checking the existence of a float in a list of 1,000,0000 strings, it takes 2-6 seconds with the current code and 1-2 seconds in the master branch. This code can be slightly optimized by storing the converted values in a type->object map. With that we see closer to the 2 second range. For a list of 100,000 strings, the difference is about 400 ms vs 300 ms. It seems to me that there isn't a huge difference until we reach very large scales. What do you think we should do?